pydm/tests/widgets/test_pushbutton.py

```diff
@@ -0,0 +1,566 @@
+        Window for widget testing
+    label : str
+        The name of the button widget
+    press_value : int, float, str
+        The value to be sent when the button is pressed
+    relative : bool
+        Send a relative value (added with the existing value) if True, or just the new value otherwise
+    init_channel : str
+        The Channel ID to send the value to
+    icon_font_name : str
+        The name of a character from a font set to use as an icon (currently using "fontawesome.ttf")
+    icon_color : QColor
+        The RGB color value for the icon
+    """
+    icon = None
+    if icon_font_name:
+        icon = IconFont().icon(icon_font_name, icon_color)
+
+    pydm_pushbutton = PyDMPushButton(label=label, pressValue=press_value, relative=relative,
+                                     init_channel=init_channel, icon=icon)
+    pydm_pushbutton.icon()
+    qtbot.addWidget(pydm_pushbutton)
+
+    assert pydm_pushbutton.text() == label if label else pydm_pushbutton.text() == ""
+    assert pydm_pushbutton.relativeChange == relative
+
+    if press_value:
+        assert pydm_pushbutton.pressValue == str(press_value)
+    else:
+        assert pydm_pushbutton.pressValue == "None" if press_value is None else pydm_pushbutton.pressValue == ""
+
+    if init_channel:
+        assert pydm_pushbutton.channel == init_channel
+    else:
+        assert pydm_pushbutton.channel == "None" if init_channel is None else pydm_pushbutton.channel == ""
+
+    if icon:
+        size = QSize(30, 30)
+        icon_pixmap = icon.pixmap(size)
+        button_icon_pixmap = pydm_pushbutton.icon().pixmap(size)
+        assert icon_pixmap.toImage() == button_icon_pixmap.toImage()
+
+    assert pydm_pushbutton.showConfirmDialog == False
+    assert pydm_pushbutton.confirmMessage == PyDMPushButton.DEFAULT_CONFIRM_MESSAGE
+    assert pydm_pushbutton.passwordProtected == False
+    assert pydm_pushbutton.password == ""
+    assert pydm_pushbutton.protectedPassword == ""
+
+
+@pytest.mark.parametrize("password_is_protected", [
+    (True),
+    (False),
+])
+def test_password_protected(qtbot, password_is_protected):
+    """
+    Test that the password protected property is properly set. The password protected is flag that, if set to True,
+    will prompt the user to enter a password before the button can be pushed to send the data
+
+    Expectation:
+    The protect password property flag retains the same Boolean value as provided.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        Window for widget testing
+
+    password_is_protected : bool
+        The flag to set the password to protected (True), or not (otherwise)
+    """
+    pydm_pushbutton = PyDMPushButton()
+    qtbot.addWidget(pydm_pushbutton)
+
+    pydm_pushbutton.passwordProtected = password_is_protected
+    assert pydm_pushbutton._password_protected == password_is_protected
+
+
+@pytest.mark.parametrize("relative_choice", [
+    (True),
+    (False),
+    (False),
+    (True),
```

Comment:
Am I misunderstanding or is this redundant?

Reply:
You are right! @hmbui, since you are always creating a new PushButton you are not testing the change of state... so just one True/False case is enough.

Reply:
I was testing state changes :-) OK, I'll keep just one True/False case pair.